spec/formatting.md

```diff
@@ -400,6 +417,23 @@ _Option_ _identifiers_ and values are not included in the _fallback value_.
 
 _Pattern selection_ is not supported for _fallback values_.
```

Comment:
Should this be optional?  `*` might match?

Reply:
If a selector does not support pattern selection, then the `*` variant is selected. A selector is never asked about `*`, that's handled by the selection algorithm directly.

The fact that you need to consider the combined effects of Resolve Preferences 2.ii.b, Filter Variants 2.i.b, and Sort Variants 5.iii.c to see that is what I tried to mean by arguing that "best choice" is too complicated.

Reply:
I went to the context and read the section carefully. I guess I have a few questions.

This text suggests that the fallback state is transitive--that implementations must track if a fallback has occurred. That probably needs to be stated more clearly, if so. Otherwise, the fallback value looks just like a string i.e. a literal, which is what lead to my comment.

Let's use my bad example:

```
.local $a = {0xbeef :num}       // this is Bad Operand with a fallback of `|0xbeef|`
.local $b = {$a :uppercase}     // this might work on `0xbeef`? It's probably Bad Operand if the first line isn't
.match $b                       // let's say :uppercase is a type of string selector too...
48879 {{Definitely not this}}
0xBEEF {{Matches?}}
* {{Could match if above doesn't}}
```

My point is, how does `.match $b` know that its resolved value is a fallback value?

Reply:
That's a good point. Some of the current text is using "fails to resolve" in a way that's not clear. I'll apply some updates to improve this.